app/src/main/java/com/aacademy/homework/data/local/MockRepository.kt

```diff
@@ -1,80 +1,36 @@
-                Actor(
-                    id = 1,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                ),
-                Actor(
-                    id = 2,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                ),
-                Actor(
-                    id = 3,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                ),
-                Actor(
-                    id = 4,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                ),
-                Actor(
-                    id = 5,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                ),
-                Actor(
-                    id = 6,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                )
-            )
-        ),
-        Movie(
-            id = 1,
-            title = "Tenet",
-            coverPath = "https://cagrikayit.com/wp-content/uploads/2020/05/tenet-vizyonda-768x961.png",
-            ageLimit = 10,
-            tags = listOf("Action", "Thriller", "Sci-Fi"),
-            rating = 4,
-            reviews = 64,
-            storyline = "Empty",
-            min = 134,
-            isLiked = false,
-            cast = listOf(
-                Actor(
-                    id = 7,
-                    firstName = "Johnny",
-                    lastName = "Depp",
-                    photoPath = "https://i.pinimg.com/236x/d8/ed/b5/d8edb51aa51788b9d3c0360acd6f345a--johnny-depp-blow-heres-johnny.jpg"
-                )
-            )
-        )
-    )
+    private val database: AppDatabase by lazy {
+        Room.databaseBuilder(MyApp.INSTANCE, AppDatabase::class.java, "sqlite.db")
+            .createFromAsset("sqlite.db")
+            .build()
+    }
+
+    fun getAllMoviePreviews(): Observable<List<MoviePreviewWithTags>> {
+        return database.movieDao().getAllMovies()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
```

Comment:
I do not recommend you to switch threads in repository. You do not know in the implementation how this repo will be used.  Switch threads where you start your rx chain